Mathlib/Tactic/TacticAnalysis.lean

```diff
@@ -0,0 +1,371 @@
+/-
+Copyright (c) 2025 Lean FRO LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+
+import Batteries.Data.Array.Merge
+import Lean.Elab.Tactic.Meta
+import Lean.Util.Heartbeats
+import Lean.Server.InfoUtils
+-- Import this linter explicitly to ensure that
+-- this file has a valid copyright header and module docstring.
+import Mathlib.Tactic.Linter.Header
+
+/-! # Tactic analysis framework
+
+In this file we define a framework for analyzing sequences of tactics.
+This can be used for linting (for instance: report when two `rw` calls can be merged into one),
+but it can also be run in a more batch-like mode to report larger potential refactors
+(for instance: report when a sequence of three or more tactics can be replaced with `grind`).
+
+## Using the framework
+
+The framework runs, but does nothing by default (`set_option linter.tacticAnalysis false`
+to turn it off completely). Enable specific analysis rounds by enabling their corresponding options:
+`set_option linter.tacticAnalysis.roundName true`.
+
+To add a round of analysis, make a definition of type `Mathlib.TacticAnalysis.Config`,
+give it the `@[tacticAnalysis linter.tacticAnalysis.roundName]` attribute, and don't forget to
+enable the option.
+
+## Warning
+
+The `ComplexConfig` interface doesn't feel quite intuitive and flexible yet and should be changed
+in the future. Please do not rely on this interface being stable.
+-/
+
+open Lean Elab Term Command Linter
+
+/-- The tactic analysis framework hooks into the linter to run analysis rounds on sequences
+of tactics.
+This can be used for linting, or in a more batch-like mode to report potential refactors.
+-/
+register_option linter.tacticAnalysis : Bool := {
+  defValue := true
+  descr := "enable the tactic analysis framework"
+}
+
+namespace Lean.Elab.ContextInfo
```

Comment:
Are these sufficiently specific to this framework that they should live in this file? What about putting them in `Mathlib/Lean/...`?